src/mod/server.mod/servmsg.c

```diff
@@ -1075,15 +1118,25 @@ static void server_activity(int idx, char *msg, int len)
   code = newsplit(&msg);
   if (raw_log && ((strcmp(code, "PRIVMSG") && strcmp(code, "NOTICE")) ||
       !match_ignore(from))) {
-    if (!strcmp(from, ""))
-      putlog(LOG_RAW, "*", "[@] %s %s", code, msg);
-    else
-      putlog(LOG_RAW, "*", "[@] %s %s %s", from, code, msg);
+      rawlen = egg_snprintf(s, sizeof s, "[@] ");
+      if (tag) {
+        rawlen += egg_snprintf(s + rawlen, sizeof s - rawlen, "%s ", tag);
+      }
+      if (strcmp(from, "")) {
+        rawlen += egg_snprintf(s + rawlen, sizeof s - rawlen, "%s ", from);
+      }
+      egg_snprintf(s + rawlen, sizeof s - rawlen, "%s %s", code, msg);
+      putlog(LOG_RAW, "*", "%s", s);
   }
   /* This has GOT to go into the raw binding table, * merely because this
    * is less efficient.
    */
-  check_tcl_raw(from, code, msg);
+//  check_tcl_raw(from, code, msg);
+  if (msgtag && tag) {
+    check_tcl_rawt(from, code, msg, tag);
```

Comment:
Could we convert this into a key/value pair list for Tcl instead? values can be normalized to "" if there is no value according to the spec and would fit nicely with Tcl

Reply:
Done- handling tags (not key/values) requires a catch on the Tcl side. Not sure if this is possible to accomplish with the API as it stands?